workspace/workflows/jobs.py

```diff
@@ -220,6 +220,15 @@ def _summarise(header_text: str, locations: list[str], skip_successful: bool) ->
     unsorted = {}
     for location in locations:
         wf_conclusions = RepoWorkflowReporter(location).get_latest_conclusions()
+
+        # Skip reporting a failure that is already known
+        known_failure_ids = config.WORKFLOWS_KNOWN_TO_FAIL.get(location, [])
+        wf_conclusions = {
+            k: v
+            for k, v in wf_conclusions.items()
+            if v == "success" or int(k) not in known_failure_ids
```

Comment:
Seeing values being converted between types in the middle of a function like this is a bit of a code smell to me.  In this case, it suggests we're not clear whether the workflow IDs should be ints or (presumably) strs, which means there's a risk we'll get them mixed up.  Ideally, we'd do any necessary conversion at the boundary of a system -- here this would either involve converting str->int in `RepoWorkflowReporter`, or converting str->int when loading `WORKFLOWS_KNOWN_TO_FAIL`, or insisting that `WORKFLOWS_KNOWN_TO_FAIL` contains strs.

No need to fix now if it's not straightforward, but something to bear in mind in future.

Reply:
This `int()` is not actually needed - I've removed it. Workflow IDs are int except when read/writing to/from the cache (due to `json.dumps`) - this is me forgetting that. I'll do some tidying in a later PR so that type conversions are only at the boundary of the cache.